### PR TITLE
Feature/jcs/bug squash profile

### LINKED ISSUE #62: Add additional FE tests
Attempt to add (not required to get a 3):
- 1 action creator test
- 1 integration test

### LINKED ISSUE #71: User actions test
### Feature Categories:

- [ ] Express
- [x] React
- [ ] Styling/CSS
- [ ] Documentation
- [x] Other: Jest Testing

**Describe the feature/Bug your gonna work on**
Add testing suite to actions in user/api/actions

**Additional context**
Add any other context or screenshots about the feature.

### LINKED ISSUE #78: User ProfileEdit form validation/error handling
### Feature Categories:
Add requirements to form fields.
- Email must be an email (no auth) just follow email structure conventions
- zip code must be a in zip code format
- Prefill form with current email and name 
- All fields must be filled

- [ ] Express
- [x] React
- [ ] Styling/CSS
- [ ] Documentation
- [ ] Other: 

**Describe the feature/Bug your gonna work on**
Adding pattern to inputs with regex pattern to check if data is allowed

**Additional context**
Add any other context or screenshots about the feature.

### LINKED ISSUE #79: User Profile Update Redirect to Profile
### Feature Categories:

- [ ] Express
- [x] React
- [ ] Styling/CSS
- [ ] Documentation
- [ ] Other: 

**Describe the feature/Bug your gonna work on**

After a successful user update redirect to user profile view so user can see updated data.

**Additional context**
Add any other context or screenshots about the feature.

### PR DESCRIPTION
**Pull request by JoshSevy**
# Crate Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Bug fixes for profile edit page
- Form validation using pattern with reger
- action tests added
- some small styling adjustments to match wireframe

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:
`Your message here`
## Has This Been Tested?
- [ ] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
npm test
# Message for reviewer:
Just fixing some bugs on the profile section of the project
# Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: code/web/src/modules/user/EditProfile.js [hunk 1]
```diff
@@ -32,8 +32,8 @@ class EditProfile extends Component {
     this.state = {
       isLoading: false, 
       newProfileData: {
-        name: '',
-        email: '',
+        name: this.props.user.details.name || "",
```
Comment (rwilliams659) [inline code/web/src/modules/user/EditProfile.js:35]: Great thought here to auto-fill the value as the name/email from the store

File: code/web/src/modules/user/EditProfile.js [hunk 2]
```diff
@@ -187,6 +181,8 @@ class EditProfile extends Component {
 
               {/* Email */}
               <Input
+                id="email-input"
+                pattern="(?!email@aol\.com$)[a-z0-9._%+-]{3,}@[a-z]{3,}\.[a-z]{2,}(?:\.[a-z]{2,})?"
```
Comment (rwilliams659) [inline code/web/src/modules/user/EditProfile.js:185]: Love the regex pattern! So great!

File: code/web/src/modules/user/api/actions.test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import configureMockStore from 'redux-mock-store'
```
Comment (mlynch5187) [inline code/web/src/modules/user/api/actions.test.js:1]: Beautiful work front end!

File: code/web/src/modules/user/Profile.js [hunk 4]
```diff
@@ -57,15 +57,17 @@ const Profile = ({ user, logout }) => {
       <Grid>
         <GridCell style={{ padding: "2em", textAlign: "center" }}>
           <img
-            src={details.image ? `${APP_URL_API}/images/uploads/${details.image}` : `${APP_URL}//images/stock/men/4.jpg`}
+            src={(details.image !== null) ?`${APP_URL_API}/images/uploads/${details.image}` : `${APP_URL}/images/stock/men/4.jpg` }
```
Comment (rwilliams659) [inline code/web/src/modules/user/Profile.js:60]: I don't think this change was necessary, but it should still work the same either way

**Review by mlynch5187 — APPROVED** [review]

File: code/web/src/modules/user/Profile.js [hunk 5]
```diff
@@ -57,15 +57,17 @@ const Profile = ({ user, logout }) => {
...
             style={{
               marginBottom: "2em",
               height: "24em",
               boxShadow: level1,
               borderRadius: "10px",
+              maxWidth: "250px",
             }}
           />
-          <H4>{details.name}</H4>
+          <H4 style={{textTransform: "capitalize"}}>{details.name}</H4>
```
Comment (rwilliams659) [inline code/web/src/modules/user/Profile.js:69]: Great thought to add this in so the name is always capitalized

File: code/web/src/modules/user/api/actions.test.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import fetchMock from 'fetch-mock'
+import thunk from 'redux-thunk'
+import * as actions from './actions'
+
+
+const middlewares = [thunk] 
+const mockStore = configureMockStore(middlewares)
+
+describe('User Actions', () => {
```
Comment (rwilliams659) [inline code/web/src/modules/user/api/actions.test.js:10]: Thanks for working on action tests!!

**Review by rwilliams659 — APPROVED** [review]
Nice work, thanks for getting some of those bug fixes & action testing done

File: code/web/src/modules/user/EditProfile.js [hunk 7]
```diff
@@ -32,8 +32,8 @@ class EditProfile extends Component {
     this.state = {
       isLoading: false, 
       newProfileData: {
-        name: '',
-        email: '',
+        name: this.props.user.details.name || "",
+        email: this.props.user.details.email || "",
```
Comment (Abdeboskey) [inline code/web/src/modules/user/EditProfile.js:36]: So simple, yet so clever!

File: code/web/src/modules/user/EditProfile.js (hunk 2 above)
Comment (Abdeboskey) [inline code/web/src/modules/user/EditProfile.js:185]: 🤤

File: code/web/src/modules/user/EditProfile.js [hunk 8]
```diff
@@ -278,6 +274,7 @@ class EditProfile extends Component {
               {/* Zip code */}
               <Input
                 type="text"
+                pattern="^\d{5}(?:[-\s]\d{4})?$"
```
Comment (Abdeboskey) [inline code/web/src/modules/user/EditProfile.js:277]: So great to know about this attribute.

File: code/web/src/modules/user/Profile.js (hunk 4 above)
Comment (Abdeboskey) [inline code/web/src/modules/user/Profile.js:60]: I think we can remove the `!== null` in this condition?

File: code/web/src/modules/user/api/actions.test.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      {isLoading: true, type: actions.LOGIN_REQUEST},
+      {error: "Please try again", type: actions.LOGIN_RESPONSE}
+    ]
+
+    return store.dispatch(actionCall).then(() => {
+      // return of async actions
+    expect(store.getActions()).toEqual(expectedActions)
+    })
+  })
+})
```
Comment (Abdeboskey) [inline code/web/src/modules/user/api/actions.test.js:48]: Excellent coverage, these tests seem quite tricky!

**Review by Abdeboskey — APPROVED** [review]
Way to go!!! We're moving right along.

**Pull request merged**